### DISCUSSION TITLE
How can I run Zed natively on wayland?

### DISCUSSION THREAD

**Discussion by azizsayfiddin** [zed-industries/zed#10601] [Feature Requests]
I installed [zed-git](https://aur.archlinux.org/packages/zed-editor-git), but it runs in xwayland instead of wayland. How can I run in wayland?

**Comment by ALEZ-DEV** [comment]
Based on what I see on the documentation and the source code, you have to set `WAYLAND_DISPLAY` variable environment to any value, there is an exemple :
clone the project -> `git clone git@github.com:zed-industries/zed.git`
cd to the project -> `cd zed`
compile the project -> `cargo build --release` (only compile because it will take a loooong time)
then run the project -> `WAYLAND_DISPLAY='true' cargo run --release`

I don't know if those command will work because zed run on wayland by default on my machine, hope this help!

**Reply by azizsayfiddin** [reply]
I thought it was running in xwayland because the cursor was blurred. It runs under wayland but the cursor is blurred.

**Reply by ALEZ-DEV** [reply]
I think that's more of a driver issue, but not sure, because the GUI framework (GPUI) is not fully supported on linux

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
